code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -207,103 +290,111 @@ GLOBAL_LIST_EMPTY(parasites) //all currently existing/living guardians
 	to_chat(src, span_notice("Your new name [span_name("[new_name]")] anchors itself in your mind."))
 	fully_replace_character_name(null, new_name)
 
-/mob/living/simple_animal/hostile/guardian/Life(delta_time = SSMOBS_DT, times_fired) //Dies if the summoner dies
-	. = ..()
-	update_health_hud() //we need to update all of our health displays to match our summoner and we can't practically give the summoner a hook to do it
-	med_hud_set_health()
-	med_hud_set_status()
-	if(!QDELETED(summoner))
-		if(summoner.stat == DEAD)
-			forceMove(summoner.loc)
-			to_chat(src, span_danger("Your summoner has died!"))
-			visible_message(span_danger("<B>\The [src] dies along with its user!</B>"))
-			summoner.visible_message(span_danger("<B>[summoner]'s body is completely consumed by the strain of sustaining [src]!</B>"))
-			for(var/obj/item/W in summoner)
-				if(!summoner.dropItemToGround(W))
-					qdel(W)
-			summoner.dust()
-			death(TRUE)
-			qdel(src)
-	else
-		to_chat(src, span_danger("Your summoner has died!"))
-		visible_message(span_danger("<B>[src] dies along with its user!</B>"))
-		death(TRUE)
-		qdel(src)
-	snapback()
+/mob/living/simple_animal/hostile/guardian/proc/on_summoner_death(mob/living/source)
+	SIGNAL_HANDLER
+
+	cut_summoner()
+	forceMove(source.loc)
+	to_chat(src, span_danger("Your summoner has died!"))
+	visible_message(span_bolddanger("\The [src] dies along with its user!"))
+	source.visible_message(span_bolddanger("[source]'s body is completely consumed by the strain of sustaining [src]!"))
+	source.dust(drop_items = TRUE)
+	death(TRUE)
+
+/mob/living/simple_animal/hostile/guardian/proc/on_summoner_deletion(mob/living/source)
+	SIGNAL_HANDLER
+
+	cut_summoner()
+	to_chat(src, span_danger("Your summoner is gone!"))
+	qdel(src)
 
 /mob/living/simple_animal/hostile/guardian/get_status_tab_items()
 	. += ..()
 	if(summoner)
-		var/resulthealth
-		if(iscarbon(summoner))
-			resulthealth = round((abs(HEALTH_THRESHOLD_DEAD - summoner.health) / abs(HEALTH_THRESHOLD_DEAD - summoner.maxHealth)) * 100)
-		else
-			resulthealth = round((summoner.health / summoner.maxHealth) * 100, 0.5)
-		. += "Summoner Health: [resulthealth]%"
-	if(cooldown >= world.time)
-		. += "Manifest/Recall Cooldown Remaining: [DisplayTimeText(cooldown - world.time)]"
+		var/resulthealth = (iscarbon(summoner) ? (abs(HEALTH_THRESHOLD_DEAD - summoner.health) / abs(HEALTH_THRESHOLD_DEAD - summoner.maxHealth)) : (summoner.health / summoner.maxHealth)) * 100
```

Comment:
can this be an if please this is a lot of logic to hold in your head on one line

Reply:
i made it a define

Reply:
nevermind its a proc now